### ISSUE TITLE
Automatic Restart After Config Change

### ISSUE THREAD

**Issue by Acmion** [sveltejs/vite-plugin-svelte#210]
### Describe the problem

When changing `svelte.config.js` during `dev` runtime, one gets an error message:
```
[vite-plugin-svelte] Svelte config change detected, restart your dev process to apply the changes.
```

### Describe the proposed solution

Create an option in `svelte.config.js` that enables automatic restart.

This is not a new idea. For example, C# (dotnet) support this. See "rude edit" [here](https://devblogs.microsoft.com/dotnet/update-on-net-hot-reload-progress-and-visual-studio-2022-highlights/#improved-user-experience-in-visual-studio-2022).

### Alternatives considered

Maybe this should be added to `vite-plugin-svelte` instead?

### Importance

nice to have

**Comment by bluwy** [comment]
Yes, this should be an issue in vite-plugin-svelte. The reason we didn't do automatic restarts was that there could be a possibility that something's not working properly (can't remember what exactly). So we opt for a message instead. Another reason was also that Vite's restart isn't a full complete restart and we weren't confident for it to be used in svelte kit.

Maybe it's worth revisiting this though, especially since svelte kit isn't using Vite via middleware mode and things may be more stable now.

**Comment by dominikg** [comment]
Code responsible  is here https://github.com/sveltejs/vite-plugin-svelte/blob/c086c9f7db99b87a53dcb04e8d07d8ee7846798b/packages/vite-plugin-svelte/src/utils/watch.ts#L47

problem is that there is no vite config file in kit so we don't have a way to trigger a vite restart.

**Comment by dominikg** [comment]
If vite exposed a restart function on devserver, we could though. I suggested it a while back here https://github.com/vitejs/vite/pull/3182#issuecomment-886474044

**Comment by dominikg** [comment]
good news. ViteDevServer restart was merged so this can be implemented when vite 2.7 is released
